dynamo-adapter.js

```diff
@@ -0,0 +1,145 @@
+const {promisify} = require('util');
+const {DynamoTable} = require('./slim');
+
+const isCallback = (fn) => typeof fn === 'function';
+
+const validateCall = (methodName, args, minLength) => {
+  if (args.length < minLength)
+    throw new Error(`Incorrect args supplied to method ${methodName}: expected: ${minLength} got: ${args.length}`);
+}
+
+/**
+  Provides an adapter for the DynamoTable class to allow for promisified
+  api usage when omitting the cb parameter.
+*/
+class DynamoAdapter {
+  constructor(name, options) {
+    this._delegate = new DynamoTable(name, options);
+    this.hooks = options.hooks || {};
+
+    // Delegate externally accessed properties
+    Object.keys(this._delegate).forEach(key => {
+      this[key] = this._delegate[key];
+    });
+
+    // NOTE: Rebind all methods so they can be passed without the class instance
+    // Since we only instantiate once per table and maintain the instances,
+    // the performance should be a big concern
+    Object.getOwnPropertyNames(DynamoAdapter.prototype).forEach(key => {
```

Comment:
can we just bind `_call` here since all of them delegate to it anyway?

Reply:
The functions need this binding to find `_call`